src/Workleap.ComponentModel.DataAnnotations/ContainsOnlyNonEmptyStringsAttribute.cs

```diff
@@ -0,0 +1,23 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+
+namespace Workleap.ComponentModel.DataAnnotations;
+
+[AttributeUsage(AttributeTargets.Property | AttributeTargets.Field | AttributeTargets.Parameter)]
+public sealed class ContainsOnlyNonEmptyStringsAttribute : ValidationAttribute
+{
+    internal const string ErrorMessageFormat = "The field {0} must be a collection that contains only non-empty Strings";
+    
+    public ContainsOnlyNonEmptyStringsAttribute() : base(ErrorMessageFormat)
+    {
+    }
+
+    public override bool IsValid(object? value) => value switch
+    {
+        null => true,
+        IEnumerable<string> enumerable => enumerable.All(x => !string.IsNullOrWhiteSpace(x)),
```

Comment:
Would we want to check for IEnumerable<string?>? Even if it's nullable maybe none are.

Reply:
That would be unreachable, it equates string with string? and binds to the first one in the switch, the compiler won't even allow it

Reply:
To be safer, for the compiler, you could use

```suggestion
        IEnumerable<string?> enumerable => enumerable.All(x => !string.IsNullOrWhiteSpace(x)),
```